src/internal/connector/onedrive/data_collections_test.go

```diff
@@ -97,13 +108,13 @@ func (suite *DataCollectionsUnitSuite) TestMigrationCollections() {
 
 				for i, cm := range test.expectMigration {
 					if cm.full == fp && cm.prev == pp {
-						migrs[i] = true
+						migrs[i] = cm
 					}
 				}
 			}
 
 			for i, m := range migrs {
-				assert.Truef(t, m, "expected to find migration: %+v", test.expectMigration[i])
+				assert.Contains(t, migrs, m, "expected to find migration: %+v", test.expectMigration[i])
 			}
```

Comment:
actually since I missed that the expected value is a slice this whole loop could be further simplified into just
```go
assert.Subset(t, migrs, test.expectMigration)
```

Or we could skip the len check above and just use `ElementsMatch` I think

Reply:
second though, Len check above is needed but can exit after it if the expected len is 0